src/Obj.hs

```diff
@@ -1121,3 +1121,10 @@ trueXObj = XObj (Bol True) Nothing Nothing
 -- | Dynamic 'false'.
 falseXObj :: XObj
 falseXObj = XObj (Bol False) Nothing Nothing
+
+-- | Applies an XObj transformation over all atomic XObjs in a form, retaining
+-- list and array structure.
+endoMap :: (XObj -> XObj) -> XObj -> XObj
+endoMap f (XObj (Arr xs) i t) = XObj (Arr (map f xs)) i t
+endoMap f (XObj (Lst xs) i t) = XObj (Lst (map f xs)) i t
```

Comment:
Should the internal definitions use some form of `endoMap` as well to enable traversing nested structures or is this an explicit non-goal?

Reply:
Good call, I think the more common use case is applying the function to all terminal points in the form, so it does make sense to recursively call `endoMap` over the lists, thanks!

Reply:
Does this mean that `(endoMap square [[1 2 3] [4 [5] 6]])` becomes `[[1 4 9] [16 [25] 36]]` ?

That seems quite unexpected to me... Don't we already have a walk?

Reply:
Yes that's right. We likely do already have this function, but I forget where it's defined and what it's called.

I can revert it back to the old definition, or if we know the walk function I can just remove this entirely and use that.

Reply:
I guess the walk is in dynamic code though, and this is just a helper function. I'm not sure what's the right call. Maybe rename this one to walk?!

Reply:
Done!